gammapy/scripts/analysis.py

```diff
@@ -32,14 +39,16 @@ class Analysis:
     Here are different examples on how to create an `Analysis` session class:
 
     >>> from gammapy.scripts import Analysis
-    >>> settings = {"general": {"out_folder": "myfolder"}}
+    >>> settings = {"general": {"outdir": "myfolder"}}
     >>> analysis = Analysis(settings)
     >>> analysis = Analysis()
     """
 
     def __init__(self, config=None):
         self._config = Config(config)
         self._set_logging()
+        self.datastore = DataStore()
```

Comment:
Suggest to init to None. I don't think having an empty DataStore or Observations object can help in any way, but it has the potential to have silent errors when users run and nothing happens because they have zero runs or observations.

I'm not sure what API and responsibilities we want concerning how Analysis interacts with DataStore and Observations. My guess would be that Analysis should have the observations data member or property (created from config), but doesn't need the `self.datastore`. The DataStore is just something that's used at the start to create the `Observations` object, but then never needed during analysis?

This Analysis class will quickly grow and become very complex, because it's the manager for all the analysis workflows. So we have to work hard to keep it as simple and small as possible, to keep it maintainable.

Reply:
Ok.
I've moved to `__init__` the declaration of all the instance *has-a*  properties defined by composition of other classes (datastore is not among them). I think it's good to have them listed at the beginning. 

```
        self.observations = None
        self.geom = None
        self.background_estimator = None
        self.extraction = None
```

Reply:
OK, let's see how it goes ...